task.go

```diff
@@ -330,3 +332,35 @@ func (e *Executor) startExecution(ctx context.Context, t *taskfile.Task, execute
 
 	return execute(ctx)
 }
+
+// GetTask will return the task with the name matching the given call from the taskfile.
+// If no task is found, it will search for tasks with a matching alias.
+// If multiple tasks contain the same alias or no matches are found an error is returned.
+func (e *Executor) GetTask(call taskfile.Call) (*taskfile.Task, error) {
+	// Search for a matching task
+	matchingTask, ok := e.Taskfile.Tasks[call.Task]
+	// If didn't find one, search for a task with a matching alias
+	if !ok {
```

Comment:
Early returns are preferred. Check for `if ok { return }` and move the current block to after it.

```go
if ok {
  return matchingTask, nil
}

// Rest of the code
```